app/channel.go

```diff
@@ -1130,6 +1130,48 @@ func (a *App) GetChannelMembersPage(channelId string, page, perPage int) (*model
 	}
 }
 
+func (a *App) GetChannelMembersTimezones(channelId string) (model.ChannelTimezones, *model.AppError) {
+	var membersTimezones model.ChannelTimezones
+	if result := <-a.Srv.Store.Channel().GetChannelMembersTimezones(channelId); result.Err != nil {
+		return nil, result.Err
+	} else {
+		membersTimezones = result.Data.(model.ChannelTimezones)
+	}
+
+	// Remove timezones that was not set in manualTimezone or automaticTimezone
+	// also remove duplicate timezones
+	var newMemberTimezones model.ChannelTimezones
+	for i := range membersTimezones {
+		if membersTimezones[i].Timezone["manualTimezone"] == "" && membersTimezones[i].Timezone["automaticTimezone"] == "" {
+			continue
+		} else {
+			if len(newMemberTimezones) == 0 {
+				newMemberTimezones = append(newMemberTimezones, membersTimezones[i])
+			} else {
+				timezoneToCheck := membersTimezones[i].Timezone["automaticTimezone"]
+				if membersTimezones[i].Timezone["automaticTimezone"] == "" {
+					timezoneToCheck = membersTimezones[i].Timezone["manualTimezone"]
+				}
+
+				found := false
+				for _, newMemberTimezone := range newMemberTimezones {
+					if timezoneToCheck == newMemberTimezone.Timezone["manualTimezone"] ||
+						timezoneToCheck == newMemberTimezone.Timezone["automaticTimezone"] {
+						found = true
+						break
+					}
+				}
+
+				if !found {
+					newMemberTimezones = append(newMemberTimezones, membersTimezones[i])
+				}
+			}
+		}
+	}
+
+	return newMemberTimezones, nil
```

Comment:
Maybe this should just return an array of strings containing the different timezone strings? I don't know how helpful it'd be to a client to return of some random users' timezone entries